Makefile

```diff
@@ -22,30 +22,36 @@ ENDPOINT_NAME = main.c
 SRC_SRCS = $(addprefix $(SOURCE_F)/, $(SRC_NAMES))
 ENDPOINT_SRC = $(addprefix $(SOURCE_F)/, $(ENDPOINT_NAME))
 
-OBJS = $(SRC_SRCS:.c=.o)
-ENDPOINT_OBJ = $(ENDPOINT_SRC:.c=.o)
+OBJ_F = build/
+OBJS = $(addprefix $(OBJ_F), $(SRC_NAMES:.c=.o))
+ENDPOINT_OBJ = $(OBJ_F)$(ENDPOINT_NAME:.c=.o)
 INCLUDES = -I . -I libft
 
 TEST_NAMES = main_test.c input_to_text_tree_test.c
 TEST_SRCS = $(addprefix $(TEST_F)/, $(TEST_NAMES))
-TEST_OBJS = $(TEST_SRCS:.c=.o)
+TEST_OBJS = $(addprefix $(OBJ_F), $(TEST_NAMES:.c=.o))
 TEST_FNAME = $(TEST_F)/test
 
-all: pre $(NAME)
+DIRS = $(COMMANDS_F) $(INPUT_TO_TEXT_TREE_MOCK_F)
+OBJ_DIRS = $(addprefix $(OBJ_F), $(DIRS))
+
+vpath %.c $(SOURCE_F) $(SOURCE_F)/$(COMMANDS_F) $(SOURCE_F)/$(INPUT_TO_TEXT_TREE_MOCK_F)
+
+all: pre $(OBJ_DIRS) $(NAME)
+
+$(OBJ_DIRS):
+	$(PREFIX)mkdir -p $(OBJ_DIRS)
 
 pre:
 	$(PREFIX)cd libft && make all
 
 $(NAME): $(OBJS) $(ENDPOINT_OBJ)
 	$(PREFIX)$(CC) $^ -o $@ $(LINK_FLAGS)
 
-$(OBJS): %.o: %.c
-	$(PREFIX)$(CC) $(COMPILE_FLAGS) $< -o $@ $(INCLUDES) $(MOCK_FLAG)
-
-$(ENDPOINT_OBJ): %.o: %.c
-	$(PREFIX)$(CC) $(COMPILE_FLAGS) $< -o $@ $(INCLUDES) $(MOCK_FLAG)
+$(OBJ_F)%.o: %.c
```

Comment:
These blocks look suspicious. Can't we merge them like
```$(OBJ_F)%.o: %.c $(TEST_F)/%.c
	$(PREFIX)$(CC) $(COMPILE_FLAGS) $< -o $@ $(INCLUDES)```?

Reply:
@ioannnikulin 

make[1]: *** [Makefile:92: memcheck] Aborted (core dumped)
make[1]: Leaving directory '/home/projects/minishell/repo'
make: *** [Makefile:98: fulltest] Error 2